third_party/parser/codegen/generate_diagnostics.cc

```diff
@@ -57,7 +57,7 @@ tuple<string, string> MESSAGES[] = {
     {"InvalidRegexp", "{}"},
     {"InvalidReturn", "invalid return in class/module body"},
     {"CSendInLHSOfMAsgn", "&. inside multiple assignment destination"},
-    {"CantAssignToNumparam", "cannot assign to numbered parameter {}"},
+    {"ReservedForNumparam", "{} is reserved for numbered parameter"},
```

Comment:
I don't know if this is the best error message, but I think we can fix this in a followup.

Reply:
Agreed. I used the same message Ruby uses.